error: `$ty:ty` is followed by `<`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:14:13
   |
LL |     ($ty:ty <) => (); //~ ERROR `$ty:ty` is followed by `<`, which is not allowed for `ty`
   |             ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$ty:ty` is followed by `<`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:15:13
   |
LL |     ($ty:ty < foo ,) => (); //~ ERROR `$ty:ty` is followed by `<`, which is not allowed for `ty`
   |             ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$pa:pat` is followed by `>`, which is not allowed for `pat` fragments
  --> $DIR/macro-input-future-proofing.rs:21:14
   |
LL |     ($pa:pat >) => (); //~ ERROR `$pa:pat` is followed by `>`, which is not allowed for `pat`
   |              ^ not allowed after `pat` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `|`, `if` or `in`

error: `$pa:pat` is followed by `$pb:pat`, which is not allowed for `pat` fragments
  --> $DIR/macro-input-future-proofing.rs:23:14
   |
LL |     ($pa:pat $pb:pat $ty:ty ,) => ();
   |              ^^^^^^^ not allowed after `pat` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `|`, `if` or `in`

error: `$pb:pat` is followed by `$ty:ty`, which is not allowed for `pat` fragments
  --> $DIR/macro-input-future-proofing.rs:23:22
   |
LL |     ($pa:pat $pb:pat $ty:ty ,) => ();
   |                      ^^^^^^ not allowed after `pat` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `|`, `if` or `in`

error: `$ty:ty` is followed by `-`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:26:13
   |
LL |     ($ty:ty -) => (); //~ ERROR `$ty:ty` is followed by `-`
   |             ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$b:ty` is followed by `-`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:27:19
   |
LL |     ($a:ty, $b:ty -) => (); //~ ERROR `$b:ty` is followed by `-`
   |                   ^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$ty:ty` is followed by `-`, which is not allowed for `ty` fragments
  --> $DIR/macro-input-future-proofing.rs:28:7
   |
LL |     ($($ty:ty)-+) => (); //~ ERROR `$ty:ty` is followed by `-`, which is not allowed for `ty`
   |       ^^^^^^^^ not allowed after `ty` fragments
   |
   = note: allowed there are: `,`, `{`, `[`, `=>`, `>`, `=`, `:`, `;`, `|`, `as` or `where`

error: `$a:expr` is followed by `$b:tt`, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:29:17
   |
LL |     ( $a:expr $($b:tt)* ) => { };
   |                 ^^^^^ not allowed after `expr` fragments
   |
   = note: allowed there are: `;`, `=>` or `,`

warning: `$a:expr` is followed (through repetition) by itself, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:31:9
   |
LL |     ( $($a:expr)* $($b:tt)* ) => { };
   |         ^^^^^^^ this fragment is followed by itself without a valid separator
   |
   = note: allowed there are: `;`, `=>` or `,`
help: add a valid separator for the repetition to be unambiguous, for example
   |
LL |     ( $($a:expr);* $($b:tt)* ) => { };
   |                 ^

error: `$a:expr` is followed by `$b:tt`, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:31:21
   |
LL |     ( $($a:expr)* $($b:tt)* ) => { };
   |                     ^^^^^ not allowed after `expr` fragments
   |
   = note: allowed there are: `;`, `=>` or `,`

error: `$a:expr` may be followed by `$b:tt`, which is not allowed for `expr` fragments
  --> $DIR/macro-input-future-proofing.rs:34:22
   |
LL |     ( $($a:expr),* $($b:tt)* ) => { };
   |                      ^^^^^ not allowed after `expr` fragments
   |
   = note: allowed there are: `;`, `=>` or `,`

error: aborting due to 11 previous errors

